cpp/src/kmeans/kmeans.hpp

```diff
@@ -22,124 +22,160 @@ namespace ML {
 
 namespace kmeans {
 
-enum InitMethod { KMeansPlusPlus, Random, Array };
+struct KMeansParams {
+  enum InitMethod { KMeansPlusPlus, Random, Array };
+
+  // The number of clusters to form as well as the number of centroids to
+  // generate (default:8).
+  int n_clusters = 8;
+
+  /*
+   * Method for initialization, defaults to k-means++:
+   *  - InitMethod::KMeansPlusPlus (k-means++): Use scalable k-means++ algorithm
+   * to select the initial cluster centers.
+   *  - InitMethod::Random (random): Choose 'n_clusters' observations (rows) at
+   * random from the input data for the initial centroids.
+   *  - InitMethod::Array (ndarray): Use 'centroids' as initial cluster centers.
+   */
+  InitMethod init = KMeansPlusPlus;
+
+  // Maximum number of iterations of the k-means algorithm for a single run.
+  int max_iter = 300;
+
+  // Relative tolerance with regards to inertia to declare convergence.
+  double tol = 1e-4;
+
+  // verbosity mode.
+  int verbose = 0;
+
+  // Seed to the random number generator.
+  int seed = 0;
+
+  // Metric to use for distance computation. Any metric from
+  // MLCommon::Distance::DistanceType can be used
+  int metric = 0;
+
+  // Oversampling factor for use in the k-means|| algorithm.
+  int oversampling_factor = 2;
+
+  int batch_size = 1 << 15;
+
+  bool inertia_check = false;
+};
 
 /**
- * @brief Compute k-means clustering and optionally predicts cluster index for each sample in the input.
+ * @brief Compute k-means clustering and optionally predicts cluster index for
```

Comment:
I'm not seeing how to make the labels computation optional. Can you elaborate upon this for the use?

Reply:
Its was optional before, now its not. The comments were fragments from the existing header file, its fixed now.